### PR TITLE
Add readme for deprected plugin (pipeline-model-declarative-agent)

### PR DESCRIPTION
**Pull request by zbynek**
Downstream of https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/386 (on hold until that one is merged)

### CODE REVIEW

**Comment by zbynek** [comment]
(I marked this as "draft" because the upstream PR must be merged first.)

**Review by oleg-nenashev — APPROVED** [review]

**Review by bitwiseman — APPROVED** [review]

**Comment by bitwiseman** [comment]
@oleg-nenashev  Merged!

**Comment by daniel-beck** [comment]
Are we waiting for a release before we can merge this?

**Comment by zbynek** [comment]
@daniel-beck I think this can be merged now, thanks.

**Comment by daniel-beck** [comment]
Right, we just should have a release of the plugin before we release https://github.com/jenkinsci/jenkins/pull/4073 otherwise users will be unhappy.

**Pull request merged**
